IcedTea-Web ResourceHandler uses a maximum of 2 * availableProcessors() threads which can prevent applications from starting

The `ThreadPoolExecutor` used to retrieve resources is limited to `Runtime.getRuntime().availableProcessors() * 2` threads, which combined with the `ThreadPoolExecutor.AbortPolicy`, can cause applications to fail to launch if there are more than `Runtime.getRuntime().availableProcessors() * 2` resources in the JNLP file. 

(We verified this by increasing the number of cores on a VM, after which our application started up fine)

We are using _AdoptOpenJDK Version 11.0.5 x64 on Win10 x64_. We have tested with AdoptOpenJDK Version 11.0.5 x86-32 too, with the same issue.

Would it possibly be better to specify that the `ThreadPoolExecutor.CallerRunsPolicy` be used instead of the default `ThreadPoolExecutor.AbortPolicy`?

https://github.com/AdoptOpenJDK/IcedTea-Web/blob/3cea3fccaf1cc8c0b45e946ec0cdb5fb6af70bad/core/src/main/java/net/adoptopenjdk/icedteaweb/resources/ResourceHandler.java#L33

```
net.sourceforge.jnlp.LaunchException: Fatal: Initialization Error: Could not initialize application. The application has not been initialized, for more information execute javaws from the command line.
	at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:669)
	at net.sourceforge.jnlp.Launcher.launchApplication(Launcher.java:411)
	at net.sourceforge.jnlp.Launcher.access$300(Launcher.java:69)
	at net.sourceforge.jnlp.Launcher$TgThread.run(Launcher.java:756)
Caused by: java.util.concurrent.RejectedExecutionException: Task net.adoptopenjdk.icedteaweb.resources.ResourceHandler$$Lambda$179/0x00000001002c2840@3f64268c rejected from java.util.concurrent.ThreadPoolExecutor@226a934b[Running, pool size = 8, active threads = 8, queued tasks = 0, completed tasks = 2]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2055)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:825)
	at java.base/java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1355)
	at net.adoptopenjdk.icedteaweb.resources.ResourceHandler.putIntoCache(ResourceHandler.java:64)
	at net.adoptopenjdk.icedteaweb.resources.ResourceTracker.startDownloadingIfPrefetch(ResourceTracker.java:186)
	at net.adoptopenjdk.icedteaweb.resources.ResourceTracker.addResource(ResourceTracker.java:158)
	at net.sourceforge.jnlp.runtime.JNLPClassLoader.initializeResources(JNLPClassLoader.java:686)
	at net.sourceforge.jnlp.runtime.JNLPClassLoader.<init>(JNLPClassLoader.java:347)
	at net.sourceforge.jnlp.runtime.JNLPClassLoader.createInstance(JNLPClassLoader.java:420)
	at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:492)
	at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:465)
	at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:661)
	... 3 more

```

--- a/core/src/main/java/net/adoptopenjdk/icedteaweb/resources/ResourceHandler.java
+++ b/core/src/main/java/net/adoptopenjdk/icedteaweb/resources/ResourceHandler.java
@@ -18,7 +18,7 @@ import java.util.List;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 import java.util.concurrent.Future;
-import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
@@ -31,7 +31,7 @@ class ResourceHandler {
 
     private static final Logger LOG = LoggerFactory.getLogger(ResourceHandler.class);
     private static final Executor localExecutor = new ThreadPoolExecutor(0, Runtime.getRuntime().availableProcessors() * 2,
-            10L, TimeUnit.SECONDS, new SynchronousQueue<>());
+            10L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
 
     private static final String LOCALHOST = "localhost";
     private static final String IPV4_LOOPBACK_PREFIX = "127.";

